internal/test/helpers.go

```diff
@@ -120,6 +120,10 @@ func TestRouteWithReply(t *testing.T, api http.Handler, external bool, method, p
 		return
 	}
 
+	if len(replyJSON) == 0 {
+		return
+	}
```

Comment:
When the worker reports its status it receives an empty body in the response, and then the unmarshal fails since it's not valid json. The alternative is to return `{}` instead of nothing in the worker server, which might actually be more correct.

Reply:
```diff --git a/internal/worker/server.go b/internal/worker/server.go
index 254c1421c..58ce6c824 100644
--- a/internal/worker/server.go
+++ b/internal/worker/server.go
@@ -1068,7 +1068,7 @@ func (h *apiHandlers) PostWorkerStatus(ctx echo.Context, workerIdstr string) err
                return api.HTTPErrorWithInternal(api.ErrorUpdatingWorkerStatus, err)
        }
 
-       return ctx.NoContent(http.StatusOK)
+       return ctx.JSON(http.StatusOK, map[string]string{})
 }
 
``` 

would fix this ig

Reply:
oh it would only fails because the expected json is `"{}"` :facepalm: , fixing